src/main/java/bio/terra/service/snapshot/flight/delete/SnapshotDeleteFlight.java

```diff
@@ -42,6 +47,13 @@ public SnapshotDeleteFlight(FlightMap inputParameters, Object applicationContext
             JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
 
         addStep(new LockSnapshotStep(snapshotDao, snapshotId, true));
+
+        // Lock the source dataset while deleting ACLs to avoid a race condition
+        // TODO note that with multi-dataset snapshots this will need to change
+        List<UUID> sourceDatasetIds = snapshotService.getSourceDatasetIdsFromSnapshotId(snapshotId);
```

Comment:
The connected tests are failing because the snapshot metadata may not exist in the case of multiple deletes 😞